### PR TITLE
feat(core): 'postCliContext' property allows context that cannot be overridden by the CLI

### PR DESCRIPTION
**Pull request by corymhall**
There are many places that a user can set `context` in a CDK application
and they are evaluated in a certain order which leads to certain methods
having precedence over others. I have added this order to the README.md.

Currently context passed through CLI arguments (i.e. `--context
SOMEKEY=true`) take precedence over values specified in the `context`
property when you declare a `new App({ context: { SOMEKEY: true } })`.
Ideally context you specify when you declare an app should take
precedence since that you can evaluate context in your code. For example
you would be able to do something like

```ts
// context from the CLI and from `cdk.json` are stored in the
// CDK_CONTEXT env variable
const cliContext = process.env.CDK_CONTEXT;

// determine whether to take the context passed in the CLI or not
const determineValue = process.env.PROD ? cliContext.SOMEKEY : 'my-prod-value';
new App({
  finalContext: {
    SOMEKEY: determineValue,
  },
});
```

Since we cannot change this behavior (would be breaking), this PR
introduces a new property `finalContext` which takes precedence.

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/core/README.md [hunk 1]
```diff
@@ -1076,4 +1076,72 @@ It's possible to synthesize the project with more Resources than the allowed (or
...
 Set the context key `@aws-cdk/core:stackResourceLimit` with the proper value, being 0 for disable the limit of resources.
 
+## App Context
+
+[Context values](https://docs.aws.amazon.com/cdk/v2/guide/context.html) are key-value pairs that can be associated with an app, stack, or construct.
+One common use case for context is to use it for enabling/disabling [feature flags](https://docs.aws.amazon.com/cdk/v2/guide/featureflags.html). There are several places
+where context can be specified. They are listed below in the order they are evaluated (items at the
+top take precedence over those below).
+
+- The `node.setContext()` method
```
Comment (rix0rrr) [inline packages/@aws-cdk/core/README.md:1086]: Can we separate out the list and the examples? I feel this is a bit hard to scan.

File: packages/@aws-cdk/core/README.md [hunk 2]
```diff
@@ -1076,4 +1076,72 @@ It's possible to synthesize the project with more Resources than the allowed (or
...
+```ts
+const app = new App();
+app.node.setContext('@aws-cdk/core:newStyleStackSynthesis', true);
+```
+
+- The `finalContext` prop when you create an `App`
+
+```ts
+new App({
+  finalContext: {
```
Comment (rix0rrr) [inline packages/@aws-cdk/core/README.md:1097]: * appContext
* independentContext
* unoverridableContext
* postContext
* postEnvContext
* internalContext
* postCliContext
* afterCliContext

?

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/core/README.md (hunk 2 above)
Comment (corymhall) [reply packages/@aws-cdk/core/README.md]: went with `postCliContext`

File: packages/@aws-cdk/core/README.md (hunk 1 above)
Comment (corymhall) [reply packages/@aws-cdk/core/README.md:1086]: done

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**